airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/CdcPostgresSourceTest.java

```diff
@@ -438,4 +449,102 @@ public SourceRecord sourceRecord() {
     assertNull(pctp.getHeartbeatPosition(null));
   }
 
+  @Test
+  protected void syncShouldIncrementLSN() throws Exception {
+    final int recordsToCreate = 20;
+
+    final DataSource dataSource = DataSourceFactory.create(
+        config.get(JdbcUtils.USERNAME_KEY).asText(),
+        config.get(JdbcUtils.PASSWORD_KEY).asText(),
+        DatabaseDriver.POSTGRESQL.getDriverClassName(),
+        String.format(DatabaseDriver.POSTGRESQL.getUrlFormatString(),
+            config.get(JdbcUtils.HOST_KEY).asText(),
+            config.get(JdbcUtils.PORT_KEY).asInt(),
+            config.get(JdbcUtils.DATABASE_KEY).asText()));
+
+    final JdbcDatabase defaultJdbcDatabase = new DefaultJdbcDatabase(dataSource);
+
+    final Long replicationSlotAtTheBeginning = PgLsn.fromPgString(
+        source.getReplicationSlot(defaultJdbcDatabase, config).get(0).get("confirmed_flush_lsn").asText()).asLong();
+
+    final AutoCloseableIterator<AirbyteMessage> firstBatchIterator = getSource()
+        .read(config, CONFIGURED_CATALOG, null);
+    final List<AirbyteMessage> dataFromFirstBatch = AutoCloseableIterators
+        .toListAndClose(firstBatchIterator);
+    final List<AirbyteStateMessage> stateAfterFirstBatch = extractStateMessages(dataFromFirstBatch);
+
+    final Long replicationSlotAfterFirstSync = PgLsn.fromPgString(
+        source.getReplicationSlot(defaultJdbcDatabase, config).get(0).get("confirmed_flush_lsn").asText()).asLong();
+    assertEquals(0, replicationSlotAfterFirstSync.compareTo(replicationSlotAtTheBeginning));
```

Comment:
nit : can you just to 
assertTrue(replicationSlotAfterFirstSync == replicationSlotAtTheBeginning) 

just find it easier to read. Same with assertions below.

Reply:
drive-by comment - `assertEquals(replicationSlotAtTheBeginning, replicationSlotAfterFirstSync)` ? (so if the assert fails, junit will also print the expected+actual values for easier debugging)